homeassistant/components/roku/__init__.py

```diff
@@ -100,17 +103,48 @@ def __init__(
         """Initialize global Roku data updater."""
         self.roku = Roku(host=host, session=async_get_clientsession(hass))
 
+        self.full_update_interval = timedelta(minutes=15)
+        self.full_update_required = True
+        self._unsub_full_update: Optional[CALLBACK_TYPE] = None
+
         super().__init__(
             hass, _LOGGER, name=DOMAIN, update_interval=SCAN_INTERVAL,
         )
 
     async def _async_update_data(self) -> Device:
         """Fetch data from Roku."""
+        full_update = self.full_update_required
+
         try:
-            return await self.roku.update()
+            data = await self.roku.update(full_update=full_update)
+
+            if full_update:
+                self.full_update_required = False
+                self._schedule_full_update()
```

Comment:
Isn't it enough to store the time of the last full update and compare with the current time? If it's been 15 min we do a full update, otherwise we do a regular update.

Reply:
thanks for the feedback, thats another approach i was considering, i suppose that is slightly less complicated to maintain overall.